rust/xaynet-server/src/state_machine/phases/idle.rs

```diff
@@ -175,80 +175,279 @@ where
 
 #[cfg(test)]
 mod tests {
-    use serial_test::serial;
-
     use super::*;
+
+    use std::sync::Arc;
+
+    use anyhow::anyhow;
+    use xaynet_core::common::RoundParameters;
+
     use crate::{
         state_machine::{
-            events::Event,
-            tests::{builder::StateMachineBuilder, utils},
+            coordinator::CoordinatorState,
+            events::{DictionaryUpdate, EventPublisher, EventSubscriber, ModelUpdate},
+            tests::{
+                utils::{assert_event_updated_with_id, enable_logging, init_shared, EventSnapshot},
+                CoordinatorStateBuilder,
+                EventBusBuilder,
+            },
+        },
+        storage::{
+            tests::{utils::create_global_model, MockCoordinatorStore, MockModelStore},
+            Store,
         },
-        storage::{tests::init_store, CoordinatorStorage},
     };
 
-    #[tokio::test]
-    #[serial]
-    async fn integration_round_id_is_updated_when_idle_phase_runs() {
-        let store = init_store().await;
-        let coordinator_state = utils::coordinator_state();
-        let (shared, _, event_subscriber) = utils::init_shared(coordinator_state, store);
+    fn state_and_events_from_unmask_phase() -> (CoordinatorState, EventPublisher, EventSubscriber) {
+        let state = CoordinatorStateBuilder::new().build();
 
-        let keys = event_subscriber.keys_listener();
-        let id = keys.get_latest().round_id;
-        assert_eq!(id, 0);
+        let (event_publisher, event_subscriber) = EventBusBuilder::new(&state)
+            .broadcast_phase(PhaseName::Unmask)
+            .broadcast_sum_dict(DictionaryUpdate::Invalidate)
+            .broadcast_seed_dict(DictionaryUpdate::Invalidate)
+            .broadcast_model(ModelUpdate::New(Arc::new(create_global_model(10))))
+            .build();
 
-        let mut idle_phase = PhaseState::<Idle, _>::new(shared);
-        idle_phase.process().await.unwrap();
-        idle_phase.broadcast();
+        (state, event_publisher, event_subscriber)
+    }
 
-        let id = keys.get_latest().round_id;
-        assert_eq!(id, 1);
+    fn assert_params(params1: &RoundParameters, params2: &RoundParameters) {
+        assert_ne!(params1.pk, params2.pk);
+        assert_ne!(params1.seed, params2.seed);
+        assert!((params1.sum - params2.sum).abs() < f64::EPSILON);
+        assert!((params1.update - params2.update).abs() < f64::EPSILON);
```

Comment:
checking for `< EPSILON` essentially means `== 0` (not exactly true, but close enough), with `<= EPSILON` we should be on the safe side.
```suggestion
        assert!((params1.sum - params2.sum).abs() <= f64::EPSILON);
        assert!((params1.update - params2.update).abs() <= f64::EPSILON);
```

similarly below.